src/app/component/ProductCard/ProductCard.component.js

```diff
@@ -74,6 +74,28 @@ export class ProductCard extends PureComponent {
 
     imageRef = createRef();
 
+    isConfigurableProductOutOfStock(productVariants) {
+        const variantsInStock = productVariants.filter((productVariant) => productVariant.stock_status === 'IN_STOCK');
+
+        if (variantsInStock.length === 0) {
+            return true;
+        }
+
+        return false;
+    }
+
+    isBundleProductOutOfStock(item) {
+        const { options } = item;
+
+        const optionsInStock = options.filter((option) => option.product.stock_status === 'IN_STOCK');
+
+        if (optionsInStock.length === 0) {
+            return true;
+        }
+
+        return false;
+    }
+
```

Comment:
Please have values like 'IN_STOCK' placed to component's config file.
Also you could have just a single return without the conditional statement, returning the comparison result or double negated length value, since these both methods gives boolean.